src/compile/data/stack.ts

```diff
@@ -21,35 +21,53 @@ function getStackByFields(model: UnitModel): string[] {
 }
 
 export interface StackComponent {
+
   /**
-   * Faceted field.
+   * Faceted field. Used in makeFromEncoding.
    */
-  facetby: string[];
+  facetby?: string[];
 
-  dimensionFieldDef: FieldDef<string>;
+  dimensionFieldDef?: FieldDef<string>;
 
   /**
-   * Stack measure's field
+   * Stack measure's field. Used in makeFromEncoding.
    */
-  field: string;
+  stackField: string;
 
   /**
    * Level of detail fields for each level in the stacked charts such as color or detail.
+   * Used in makeFromEncoding.
    */
-  stackby: string[];
+  stackby?: string[];
 
   /**
    * Field that determines order of levels in the stacked charts.
+   * Used in both but optional in transform.
    */
   sort: VgSort;
 
-  /** Mode for stacking marks. */
+  /** Mode for stacking marks.
+   */
   offset: StackOffset;
 
   /**
-   * Whether to impute the data before stacking.
+   * Whether to impute the data before stacking. Used only in makeFromEncoding.
    */
-  impute: boolean;
+  impute?: boolean;
+
+  /**
+   * The data fields to group by.
+   */
+  groupby?: string[];
+  /**
+   * Output field names of each stack field.
+   */
+  as: string[];
+
+}
+// TODO Better Name for this function
+function isAsValidArray(as: string[] | string): as is string[] {
+  return isArray(as) && as.every(s => isString(s)) && as.length ===2;
```

Comment:
If you really wanna keep this method, check if array.length > 1.  (The 3rd, 4th, ... elements can be ignored.)